src/lib/y2storage/proposal/settings_adjustment.rb

```diff
@@ -0,0 +1,126 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Storage
+  module Proposal
+    # Class to represent the different adjustments in the default proposal
+    # settings performed by each attempt during {GuidedProposal.initial}, so
+    # that information can be kept and displayed to the user.
+    #
+    # @note Although the name of the class and methods are designed to support
+    #   more scenarios in the future, the current implementation only supports
+    #   disabling attributes in concrete volumes, since that's the only action
+    #   that {GuidedProposal.initial} performs so far.
+    class SettingsAdjustment
+      include Yast::I18n
+      include Yast::Logger
+
+      def initialize
+        textdomain "storage"
+        @volumes = {}
+      end
+
+      # New adjustment object created by registering a new modification to the
+      # settings of a concrete volume
+      #
+      # @param volume [VolumeSpecification] concrete volume that was configured
+      # @param attribute [Symbol] attribute of the volume that was adjusted
+      # @param value [Object] value set for the attribute
+      #
+      # @return [SettingsAdjustment]
+      def add_volume_attr(volume, attribute, value)
+        log.debug "Adjustments: add volume #{volume.mount_point} with #{attribute} == #{value}"
+        result = dup
+        result.volumes[volume.mount_point] ||= []
+        result.volumes[volume.mount_point] << attribute
+        log.debug "New adjustments: #{result.volumes}"
+        result
+      end
+
+      # User-oriented localized sentences explaining each aspect of the
+      # adjustment that have been performed on top of the default settings
+      #
+      # @return [Array<String>]
```

Comment:
please mention that volumes is returned sorted or it is just implementation detail?

Reply:
I just wanted to ensure some kind of consistency since the strings will be displayed to the user, but the current sorting (by volume path) is actually just an arbitrary criterion. Nothing so relevant or well-though to be documented, IMHO.